### PR TITLE
Add pgwire tunneling for CCR

### PR DESCRIPTION
**Pull request by mfussenegger**
TODO:

- [x] Fix flakyness (? - wasn't able to reproduce anymore)
- [x] Password auth on client side
- [x] SSL on client side
- [x] Docs

I'll squash all commits down to 1 before merging. Kept some of them as it might help with the review

### CODE REVIEW

**Comment by mfussenegger** [comment]
@seut Could you have a first look at this? There are still some TODOs left but I think the overall structure should be good.

Still trying to figure out if flakyness is related to the pg-tunneling or not - unfortunately it only failed once in ~90 minutes of non-stop repeat

**Comment by seut** [comment]
@mfussenegger great stuff, had a high level look on it and it looks good to me. Or anything concrete where I should review more detailed already?

File: server/src/main/java/org/elasticsearch/node/Node.java [hunk 1]
```diff
@@ -771,6 +780,22 @@ public Node(final Environment environment,
...
+                                                 NettyBootstrap nettyBootstrap) {
+        return new PgClientFactory(
+            settings,
+            transportService,
+            transport,
+            sslContextProvider,
+            pageCacheRecycler,
+            nettyBootstrap
+        );
+    }
```
Comment (seut) [inline server/src/main/java/org/elasticsearch/node/Node.java:797]: why is this needed? looks like it passes everything to the ctor.

File: server/src/test/java/io/crate/integrationtests/PgTunnelLogicalReplicationITest.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+import io.crate.protocols.ssl.SslSettings;
+import io.crate.replication.logical.LogicalReplicationSettings;
+import io.crate.testing.Asserts;
+import io.crate.testing.SQLErrorMatcher;
+import io.crate.testing.SQLTransportExecutor;
+import io.crate.testing.TestingHelpers;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.ssl.util.SelfSignedCertificate;
+
+public class PgTunnelLogicalReplicationITest extends ESTestCase {
```
Comment (seut) [inline server/src/test/java/io/crate/integrationtests/PgTunnelLogicalReplicationITest.java:61]: Whats the reason to not extend it from `LogicalReplicationTestCase`?

File: server/src/main/java/io/crate/protocols/postgres/PgClient.java [hunk 3]
```diff
@@ -0,0 +1,655 @@
...
+
+                default:
+                    throw new UnsupportedOperationException(authType + " is not supported");
+            }
+        }
+    }
+
+    enum AuthType {
+        OK,
+        KERBEROS,
```
Comment (seut) [inline server/src/main/java/io/crate/protocols/postgres/PgClient.java:444]: Is this used somewhere? (same for MD5_Password) exists to fulfill the protocol spec?

**Review by seut — APPROVED** [review]
👍 Nice, added just a few questions.

File: server/src/test/java/io/crate/integrationtests/PgTunnelLogicalReplicationITest.java (hunk 2 above)
Comment (mfussenegger) [reply server/src/test/java/io/crate/integrationtests/PgTunnelLogicalReplicationITest.java:61]: Mostly because I wanted more control over the cluster creation due to the different settings used (hba and ssl).

I could try to follow up on this to share some more logic

**Comment by mfussenegger** [comment]
retest this please

Unrelated doc-test failure: `http.client.RemoteDisconnected: Remote end closed connection without response`
